Rubberduck.VBEditor.VBA/SafeComWrappers/VB/VBComponent.cs

```diff
@@ -96,12 +96,18 @@ public bool HasDesigner
         /// </summary>
         /// <param name="folder">Destination folder for the resulting source file.</param>
         /// <param name="tempFile">True if a unique temp file name should be generated. WARNING: filenames generated with this flag are not persisted.</param>
-        /// <param name="specialCaseDocumentModules">If reimpot of a document file is required later, it has to receive special treatment.</param>
+        /// <param name="specialCaseDocumentModules">If reimport of a document file is required later, it has to receive special treatment.</param>
         public string ExportAsSourceFile(string folder, bool tempFile = false, bool specialCaseDocumentModules = true)
         {
             var fullPath = tempFile
                 ? Path.Combine(folder, Path.GetRandomFileName())
                 : Path.Combine(folder, SafeName + Type.FileExtension());
+
+            if (!Directory.Exists(folder))
+            {
+                Directory.CreateDirectory(folder);
```

Comment:
This needs to be in a `try`.

Reply:
The issue is that none of the export methods are wrapped similarly, except for a section within the `ExportUserFormModule`. I was following along the existing behavior. I think that if we want to change the behavior, that would be more than just fixing the bug?

Reply:
Well, the behavior currently (assuming this is the fix for #4454) is to make the parser fail. If the root issue is not being able to create the RD temp folder, I'm not sure how this fixes it when it's that far down the call stack.  Is there a way to ensure that it exists on startup or at the call site? It strikes me that we're trading one potential IO error for another potential IO error at a place where it isn't really ideal to handle it.

Reply:
Here's the thing - from what I see in the #4454 and from discussion, the folder did not exist. We _already_ have a startup routine that ensure it exists. How did we get to that state, I'm not clear. I have to agree with you, however, that we should not be scattering all the IO concerns all over. We should be making it the responsibility of .... `ITempSourceFileHandler`? `IRootPathProvider`? Just go "hey, write this junk to there, kthxbye" from here. I'll have a review and rethink that.